apps/passport-server/src/apis/devconnect/devconnectPretixAPI.ts

```diff
@@ -28,9 +34,96 @@ export interface IDevconnectPretixAPI {
     orgUrl: string,
     token: string
   ): Promise<DevconnectPretixEvent[]>;
+  cancelPendingRequests(): void;
+}
+
+export interface DevconnectPretixAPIOptions {
+  tokenRequestsPerMinute?: number;
+  concurrency?: number;
+  interval?: number;
 }
 
 export class DevconnectPretixAPI implements IDevconnectPretixAPI {
+  private tokenRequestsPerMinute: number;
+  private concurrency: number;
+  private interval: number;
+  private requestQueues: Map<string, PQueue>;
```

Comment:
does `PQueue` take a type parameter? If it does, we should use it.

Reply:
It does, but only for changing the internal queueing implementation, e.g. priority queue. It doesn't take a type parameter for the kinds of promise that might be placed on the queue.